src/libfmt_macros/lib.rs

```diff
@@ -77,9 +77,39 @@ pub struct FormatSpec<'a> {
     pub ty: &'a str,
 }
 
+/// For internal use; same as FormatSpec except using CountInternal instead
+/// of Count. Possible next-references in the count fields will be resolved
+/// before the struct is converted into FormatSpec for returning.
+struct FormatSpecInternal<'a> {
```

Comment:
We don't have private enum variants at the moment, and I don't want to add a lot of `_` match arms to other places where the enums are consumed. But I'll look into this later though, currently outside.